src/FastRandom.cpp

```diff
@@ -8,8 +8,11 @@
 
 #include <random>
 
-#if defined (__GNUC__)
-	#define ATTRIBUTE_TLS static __thread
+#if defined (ANDROID)
+	// Tycho: I'm sorry, I'm sorry, I'm so, so, sorry, but please blame the NDK developers for lack of, thread_local
+	#define ATTRIBUTE_TLS static
```

Comment:
Just leaving it as a static is UB so not really an option. So we either need to use atomics (is this possible on android?), the pthread thread local api (slow), or work out some way of using a mutex (slow).